pkg/controller/kubelet-config/helpers.go

```diff
@@ -184,14 +184,18 @@ func getManagedKubeletConfigKey(pool *mcfgv1.MachineConfigPool, client mcfgclien
 		if val != "" {
 			return fmt.Sprintf("99-%s-generated-kubelet-%s", pool.Name, val), nil
 		}
+		// the first kubeletconfig is being synced, return its default MC name
+		if ok && val == "" {
+			return ctrlcommon.GetManagedKey(pool, client, "99", "kubelet", getManagedKubeletConfigKeyDeprecated(pool))
+		}
 	}
 
 	// If we are here, this means that a new kubelet config was created, so we have to calculate the suffix value for its MC name
 	suffixNum := 0
 	// Go through the list of kubelet config objects created and get the max suffix value currently created
 	for _, item := range kcList {
 		val, ok := item.GetAnnotations()[ctrlcommon.MCNameSuffixAnnotationKey]
-		if ok {
+		if ok && val != "" {
```

Comment:
Wouldn't the return you added above be hit anytime `val == ""` making this condition unnecessary?